staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -815,14 +833,14 @@ type ruleCost struct {
 	cost uint64
 }
 
-// MultiplyByElementCost returns a costInfo where the MaxCardinality is multiplied by the
+// CostForChildSchema returns a costInfo where the MaxCardinality is multiplied by the
 // factor that the schema increases the cardinality of its children. If the costInfo's
 // MaxCardinality is unbounded (nil) or the factor that the schema increase the cardinality
 // is unbounded, the resulting costInfo's MaxCardinality is also unbounded.
-func (c *costInfo) MultiplyByElementCost(schema *apiextensions.JSONSchemaProps) costInfo {
+func (c *costInfo) CostForChildSchema(schema *apiextensions.JSONSchemaProps) costInfo {
 	result := costInfo{TotalCost: c.TotalCost, MaxCardinality: unbounded}
```

Comment:
the returned object here does not inherit any lazily computed Schema or DeclType from `costInfo`. I'm pretty disoriented w.r.t. the flow where this is called, but will that make us have to recompute things we expected to only need to be computed once?

Reply:
I've restructured this so that the flow is much simpler.  Each call these functions does everything needed and the `LoadSchemaIfAbsent` call has been removed.